### PR TITLE
Add health check to the upgrade test

### PR DESCRIPTION
**Pull request by petr-balogh**
Ticket:
https://projects.engineering.redhat.com/browse/OCSQE-145

Signed-off-by: Petr Balogh <pbalogh@redhat.com>

### CODE REVIEW

**Comment by petr-balogh** [comment]
Running verification here: https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/2977/

File: ocs_ci/ocs/cluster.py [hunk 1]
```diff
@@ -427,6 +431,86 @@ def check_ceph_pool_used_space(self, cbp_name):
...
+
+    def run(self):
+        self.ceph_cluster.health_monitor_enabled = True
+        while self.ceph_cluster.health_monitor_enabled and (
+            not self.ceph_cluster.health_error_status
+        ):
+            sleep(self.sleep)
+            health_status = self.ceph_cluster.get_ceph_health(detail=True)
+            if "HEALTH_ERROR" in health_status:
+                self.ceph_cluster.health_error_status = (
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py]: This means that only 1 error occurrence will be assigned to self.ceph_cluster.health_error_status IIUC.
What if we want more then 1 error occurrence to be saved?

**Comment by petr-balogh** [comment]
Verified:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/2982/console

File: ocs_ci/ocs/cluster.py (hunk 1 above)
Comment (petr-balogh) [reply ocs_ci/ocs/cluster.py]: I was thinking about adding more as well but then I realized that if I run this health check each 5 second and I will hit HEALTH_ERROR and this one will repeat all the time that I will spam this list with the same error all the time and no more additional error.  I thought about concentrating for first occurrence of HEALTH_ERROR only and stop monitoring after that, let process the rest of the upgrade which will most likely fail later on anyway and collect must gather and raise the exception with that first error.

@nehaberry or do you think that we should try to collect all the errors again and again?

Thanks

Comment (ebenahar) [reply ocs_ci/ocs/cluster.py]: @petr-balogh OK. Make sense to have it saved only once. Thanks!

File: ocs_ci/ocs/cluster.py [hunk 2]
```diff
@@ -427,6 +431,86 @@ def check_ceph_pool_used_space(self, cbp_name):
...
+                    self.ceph_cluster.ceph_status()
```
Comment (shylesh) [inline ocs_ci/ocs/cluster.py]: @petr-balogh , is that call ```self.ceph_cluster.get_ceph_status() ```?

Comment (petr-balogh) [reply ocs_ci/ocs/cluster.py]: ops, I changed the name of function with prefix get_ and forgot to change it here. Thanks for noticing. Submitting fix in a minute.

**Pull request merged**